src/main/java/touro/spf/InputManager.java

```diff
@@ -0,0 +1,92 @@
+package touro.spf;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.*;
+import java.util.regex.Pattern;
+
+/**
+ * Creates Node instances using data from input file. Passes
+ * the Nodes to graph.
+ */
+
+public class InputManager {
+
+    private final Scanner scanner;
+    private Graph graph;
+    private ArrayList<String> eachLine = new ArrayList<>();
+    private HashSet<Node> nodes = new HashSet<>();
+
+    public InputManager(String fileName, Graph graph) throws FileNotFoundException {
+        scanner = new Scanner(new FileInputStream(fileName));
+        this.graph = graph;
+        readFile();
+    }
+
+    private void readFile() {
+//        while (scanner.hasNextLine()) {
```

Comment:
This line and 64, 65 is commented out but can be used if we want to generate multiple graphs.